### PR TITLE
Fixed the issue with linked samples not showing if the file is in a subfolder

### PR DESCRIPTION
**Pull request by jdewell**
…ubfolder

### CODE REVIEW

File: src/hci/gnomex/controller/GetRequestDownloadList.java [hunk 1]
```diff
@@ -725,11 +725,14 @@ private void recurseAddFiles(Element fdNode, File f1, String requestNumber, Stri
 			fileNode.setAttribute("state", "unchecked");
 			fileNode.setAttribute("canDelete", "Y");
 			fileNode.setAttribute("canRename", "Y");
-			//fdNode.setAttribute("linkedSampleNumber", getLinkedSampleNumber(sess, fileName.substring(fileName.indexOf(Request.getBaseRequestNumber(requestNumber)))));
+			StringBuffer fname = new StringBuffer();
+			fname.append(fullPath.substring(fullPath.indexOf(Request.getBaseRequestNumber(fd.getNumber()))).replace("\\","/"));
+			fname.append(f.getName());
+			fileNode.setAttribute("linkedSampleNumber", getLinkedSampleNumber(sess, fname.toString()));
```
Comment (tmaness) [inline src/hci/gnomex/controller/GetRequestDownloadList.java]: You can get rid of fname and just use fd.getZipEntryName()

**Comment by tmaness** [comment]
Looks good to me.  I wonder who commented out that line?  I'm glad only B2B uses this feature they don't have many files. Otherwise we'd have to rethink the database structure because showing linked samples is VERY SLOW.

**Comment by tcourdy** [comment]
Hey @jdewell 

Tim just brought up a good point that I forgot to mention when I assigned this to you.  We also have a files tab with view links on the Analysis object.  Could you take a look into this instance as well?

**Comment by jdewell** [comment]
@tcourdy. Thanks for the heads up. I'll look into this.

File: src/hci/gnomex/controller/GetRequestDownloadList.java (hunk 1 above)
Comment (jdewell) [inline src/hci/gnomex/controller/GetRequestDownloadList.java]: Tim,
I found that the method getLinkedSampleNumber expects the format of experimentNumber/directory (if there is one)/name of file.

Maybe I’m missing something.

Thanks,
-John

File: src/hci/gnomex/controller/GetRequestDownloadList.java [hunk 2]
```diff
@@ -717,19 +723,22 @@ private void recurseAddFiles(Element fdNode, File f1, String requestNumber, Stri
...
+			FileDescriptor fd = new FileDescriptor(requestNumber, f.getName(), f, fName);
 			fd.setDirectoryName(directoryName);
 			fd.excludeMethodFromXML("getChildren");
 			Element fileNode = fd.toXMLDocument(null, fd.DATE_OUTPUT_ALTIO).getRootElement();
 			fileNode.setAttribute("isSelected", "N");
 			fileNode.setAttribute("state", "unchecked");
 			fileNode.setAttribute("canDelete", "Y");
 			fileNode.setAttribute("canRename", "Y");
-			//fdNode.setAttribute("linkedSampleNumber", getLinkedSampleNumber(sess, fileName.substring(fileName.indexOf(Request.getBaseRequestNumber(requestNumber)))));
+			String fname = this.getPathForZipFileName(f, requestNumber);
```
Comment (matthewbjorge) [inline src/hci/gnomex/controller/GetRequestDownloadList.java]: You don't need this second variable. Perhaps you could also move the "fName" and "fd" lines down here so everything is together. Maybe rename "fName" to "fPath" as well.

**Comment by tmaness** [comment]
Looks good.

**Comment by jdewell** [comment]
For whatever reason, the line that set the attribute 'linkedSampleNumber was commented out. When working on this I realized that the attribute zipEntryName was not being set correctly. This modification makes no visible changes. It just sets the zipEntryName correctly if someone else needs this attribute in the future.

**Pull request merged**